packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -322,17 +310,14 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
               fullscreenDialog: true,
             ),
           );
-
-          setState(() {
-            _nutritionFactsAdded = result != null;
-          });
+          setState(() {});
         },
       ),
     );
   }
 
-  Widget _buildaddInputDetailsButton() {
-    if (_basicDetailsAdded) {
+  Widget _buildAddInputDetailsButton() {
+    if (_product.productName != null && _product.productName!.isNotEmpty) {
```

Comment:
Could be simplified with  `_product.productName?.isNotEmpty == true`

Reply:
Fair enough.